master/internal/api_command.go

```diff
@@ -176,7 +177,7 @@ func (a *apiServer) GetCommands(
 		return nil, err
 	}
 
-	workspaceNotFoundErr := api.NotFoundErrs("workspace", fmt.Sprint(req.WorkspaceId), true)
+	workspaceNotFoundErr := api.NotFoundErrs("workspace", strconv.Itoa(int(req.WorkspaceId)), true)
```

Comment:
Nit: this linter feels extremely picky. If the reason it cares about this change is performance, there's almost zero chance we'll ever care about the performance difference between these two. I don't object to this linter in particular and It's not a huge deal, but I really feel like we should be defaulting to not having linters and only adopting them when there's a real problem that linter fixes/helps avoid.

```
$ go test -bench=.    
goos: darwin
goarch: arm64
pkg: github.com/determined-ai/determined/master/internal
BenchmarkConvUsingSprint-10    	20826658	        50.84 ns/op
BenchmarkConvUsingItoa-10      	68434394	        18.54 ns/op
```
from code
```go
import (
	"fmt"
	"strconv"
	"testing"
)

func convUsingFmt(someNum int32) string {
	return fmt.Sprint(someNum)
}

func convUsingItoa(someNum int32) string {
	return strconv.Itoa(int(someNum))
}

func BenchmarkConvUsingFmt(b *testing.B) {
	for i := int32(0); int(i) < b.N; i++ {
		convUsingFmt(i)
	}
}

func BenchmarkConvUsingItoa(b *testing.B) {
	for i := int32(0); int(i) < b.N; i++ {
		convUsingItoa(i)
	}
}
```
It's hard to imagine a time where we'll care about that ~32ns.

Reply:
I find `strconv.Itoa` to be more clear, especially in a simple use case like this. 

I think there's a bug-reduction benefit too. `fmt.Sprintf("%d", "string")` will compile but have a runtime error. (Most editors will warn you though). 

Using `strconv.Itoa("string")` won't even compile, so we get harder validation earlier. 

I'm ok with not using a linter to enforce it and leave it up to developers to make the call on what's best situationally. But I also think we should leave the conversions as is because it's a nice update to the codebase.

Reply:
> I find strconv.Itoa to be more clear, especially in a simple use case like this.

Understandable! That's a totally valid preference. Mine is the opposite: I find a dead simple case like this to be more readable with fewer layers and parens and (visible) conversion operations.

> I think there's a bug-reduction benefit too. fmt.Sprintf("%d", "string") will compile but have a runtime error. (Most editors will warn you though).

You're right that it builds, but it fails `go vet` (and thus `go test`) so IMO isn't enough of a risk to be worth restricting expressiveness over.

> I'm ok with not using a linter to enforce it and leave it up to developers to make the call on what's best situationally.

Great! Glad we're on the same page. This was the main point I was trying to make, here.

> But I also think we should leave the conversions as is because it's a nice update to the codebase.

I feel entirely willing to compromise on this particular case as an exception, but I'd really like to avoid PRs (or changes in PRs) that change code purely to change from one stylistic preference to another, and the work already having been done should generally not be a good enough reason.

Reply:
I wasn't thinking about `go vet` - good point. 

I'll leave it up to you and @carolinaecalderon to decide.

Reply:
Leaving this in, if someone wants to undo it, then they're free to

Reply:
>  but I'd really like to avoid PRs (or changes in PRs) that change code purely to change from one stylistic preference to another

fwiw +1 . purely based on making git history harder to follow for me (maybe there's a better way to use git that doesn't care about this)